presentation/src/main/java/com/wednesday/template/presentation/MainActivity.kt

```diff
@@ -1,93 +1,99 @@
 package com.wednesday.template.presentation
 
-import android.annotation.SuppressLint
 import android.os.Bundle
 import androidx.activity.compose.setContent
-import androidx.annotation.IdRes
-import androidx.annotation.NavigationRes
 import androidx.appcompat.app.AppCompatActivity
 import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
 import androidx.compose.foundation.layout.fillMaxSize
 import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.material.Button
 import androidx.compose.material.Scaffold
 import androidx.compose.material.Text
-import androidx.compose.ui.Alignment
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.livedata.observeAsState
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.unit.dp
-import androidx.core.os.bundleOf
 import androidx.core.splashscreen.SplashScreen.Companion.installSplashScreen
-import androidx.fragment.app.FragmentActivity
-import androidx.navigation.NavController
-import androidx.navigation.NavGraph
-import androidx.navigation.fragment.NavHostFragment
-import androidx.navigation.fragment.findNavController
-import com.wednesday.template.presentation.weather.home.HomeScreen
-import com.wednesday.template.resources.databinding.ActivityMainBinding
+import com.wednesday.template.presentation.base.extensions.asString
+import com.wednesday.template.presentation.weather.home.HomeScreenIntent
+import com.wednesday.template.presentation.weather.home.HomeViewModel
+import org.koin.androidx.viewmodel.ext.android.viewModel
+import timber.log.Timber
 
 class MainActivity : AppCompatActivity() {
-    private lateinit var binding: ActivityMainBinding
+    val viewModel: HomeViewModel by viewModel()
+
 
     override fun onCreate(savedInstanceState: Bundle?) {
         // Handle the splash screen transition.
         installSplashScreen()
 
         super.onCreate(savedInstanceState)
-        binding = ActivityMainBinding.inflate(layoutInflater)
-        displayFragment()
 
+
+
+        viewModel.onCreate(null)
```

Comment:
Why we are using this

Reply:
Usually, this is called by the `BaseFragment`, but since we don't have a fragment anymore, this never gets called. 
This function initializes the default state in the ViewModel.

Later on when we create a BasePage composable and have the navigation setup, that will handle this initialization part.